Sources/Subs-Calendar.php

```diff
@@ -1851,4 +1851,21 @@ function removeHolidays($holiday_ids)
 	));
 }
 
+/**
+ * Helper function to convert date string to english
+ * so that date_parse can parse the date
+ *
+ * @param string $date A localized date string
+ * @return string English date string
+ */
+function convertDateToEnglish($date)
+{
+	global $txt, $context;
+
+	if ($context['user']['language'] == 'english')
+		return $date;
+
+	return strtr($date, array_combine(array_merge($txt['months_titles'], $txt['months_short']), array_merge(array('January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'), array('Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'))));
```

Comment:
It now occurs to me that it would be a good idea to perform substitutions for `$txt['days']` and `$txt['days_short']` here as well, just in case.

Reply:
Done